### PR TITLE
docs: User and Dev guide updates, generate time.observation in collectors

### PR DESCRIPTION
**Pull request by sebix**
Updated User-Guide.md and Developement-Guide.md

time.observation is generated in collectors, not in parsers. Adjusted the code including the tests accordingly.

Switch parameters of utils.log: first (most important) os now name, log path is second with a default value.

### CODE REVIEW

File: intelmq/bots/parsers/hphosts/parser.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 
 from intelmq.lib import utils
 from intelmq.lib.bot import Bot
-from intelmq.lib.harmonization import DateTime, IPAddress
```
Comment (SYNchroACK) [inline intelmq/bots/parsers/hphosts/parser.py]: IPAddress is needed

File: docs/User-Guide.md [hunk 2]
```diff
@@ -81,37 +94,77 @@ usage:
...
+* `startup.conf`: Maps the bot ids to python modules.
+* `runtime.conf`: Configuration for the individual bots.
+* `pipeline.conf`: Defines source and destination queues per bot.
+* `BOTS`: Includes configuration hints for all bots. E.g. feed URLs or
+database connection parameters. Use this as a template for `startup.conf`
+and `runtime.conf`. Also read by the intelmq-manager.
+
+To configure a new BOT, you need to define it first in `startup.conf`.
+Then do the configuration in `runtime.conf` using the bot if.
+Configure source and destination queues in `pipeline.conf`.
```
Comment (SYNchroACK) [inline docs/User-Guide.md]: here we can add a note mentioning the intelmq-manager to help generating the configs needed

File: docs/User-Guide.md [hunk 3]
```diff
@@ -45,27 +46,39 @@ chown -R intelmq.intelmq /opt/intelmq
...
+## How it Works
+Intelmq has a modular structure consisting of bots. There are four types of bots:
+
+* *CollectorBots* retrieve data from internal or external sources, the output
+are *reports* consisting of many individual data sets.
+* *ParserBots* parse the data by splitting it into individual *events* and
+giving them a defined structure, see also [Data-Harmonization](Data-Harmonization.md).
+* *ExpertBots* enrich the existing events by e.g. reverse records, geographic location
+information or abuse contacts.
+* *OutputBots* write events to files or databases.
```
Comment (SYNchroACK) [inline docs/User-Guide.md]: I think we should change to the following:
```
* *OutputBots* write events to files, databases, etc...
```

File: docs/Developers-Guide.md [hunk 4]
```diff
@@ -172,107 +151,74 @@ intelmq/bots/parser/cymru_full_bogons
...
-
-#### Back-end independence
-
-Any component of the IntelMQ MUST be independent of the message queue technology (Redis, RabbitMQ, etc...), except 'lib/pipeline.py'. Intelmq bots MAY only assume to use the class specified in 'lib/pipeline.py' for inter-process or inter-bot communications.
-
-#### Compatibility
-
-IntelMQ core (including intelmqctl) MUST be compatible with IntelMQ Manager, IntelMQ UI and IntelMQ Mailer.
-
+The `exception` method automatically appends an exception traceback. The logger instance writes to the file `/opt/intelmq/var/log/[bot-id].log` and to stderr.
```
Comment (SYNchroACK) [inline docs/Developers-Guide.md]: proposal:

```
The `exception` method automatically appends an exception traceback. The logger instance writes by default to the file `/opt/intelmq/var/log/[bot-id].log` and to stderr.
```

File: docs/Developers-Guide.md [hunk 5]
```diff
@@ -282,40 +228,136 @@ Redis is used as:
 <a name="bot-developer-guide"></a>
 ## Bot Developer Guide
 
-### Template
+There's a dummy bot including tests at `intelmq/tests/bots/test_dummy_bot.py`.
 
+You can always start any parser directly from command line by either invoking the script or the python module. Don't forget to give an bot id as first argument.
```
Comment (SYNchroACK) [inline docs/Developers-Guide.md]: I thiink we should mention that run with other user than intelmq can create some problems with log permissions etc...

File: intelmq/bots/parsers/hphosts/parser.py (hunk 1 above)
Comment (sebix) [inline intelmq/bots/parsers/hphosts/parser.py]: fixed

File: docs/User-Guide.md (hunk 2 above)
Comment (sebix) [inline docs/User-Guide.md]: Added in next line.

**Comment by SYNchroACK** [comment]
Thank you @sebix

**Pull request merged**